src/hidimstat/base_perturbation.py

```diff
@@ -0,0 +1,132 @@
+import numpy as np
+import pandas as pd
+from joblib import Parallel, delayed
+from sklearn.base import BaseEstimator, check_is_fitted
+from sklearn.metrics import root_mean_squared_error
+
+from hidimstat.utils import _check_vim_predict_method
+
+
+class BasePerturbation(BaseEstimator):
+    def __init__(
+        self,
+        estimator,
+        loss: callable = root_mean_squared_error,
+        n_permutations: int = 50,
+        method: str = "predict",
+        n_jobs: int = 1,
+    ):
+        """
+        Base class for model agnostic variable importance measures based on
+        perturbation.
+
+        Parameters
+        ----------
+        estimator : object
+            The estimator to use for the prediction.
+        loss : callable, default=root_mean_squared_error
+            The loss function to use when comparing the perturbed model to the full
+            model.
+        n_permutations : int, default=50
+            Only for PermutationImportance or CPI. The number of permutations to
+            perform.
+        method : str, default="predict"
```

Comment:
Not super clear what this represents

Reply:
Hope this is clearer.

Reply:
The possible argument is "predict",  "predict_proba", "decision_function", "transform".
There are mentioned just below.